rfc/2022/2022-06-15_IET_002_TraefikEE-Keycloak-SSO-on-Kubernetes-Ingress.md

```diff
@@ -0,0 +1,306 @@
+  "ClientPort": "57398",
+  "ClientUsername": "-",
+  "DownstreamContentSize": 385,
+  "DownstreamStatus": 200,
+  "Duration": 5386497,
+  "OriginContentSize": 385,
+  "OriginDuration": 5241769,
+  "OriginStatus": 200,
+  "Overhead": 144728,
+  "RequestAddr": "traefikee-test.vsp.local",
+  "RequestContentSize": 0,
+  "RequestCount": 7,
+  "RequestHost": "traefikee-test.vsp.local",
+  "RequestMethod": "GET",
+  "RequestPath": "/whoami",
+  "RequestPort": "-",
+  "RequestProtocol": "HTTP/1.1",
+  "RequestScheme": "http",
+  "RetryAttempts": 0,
+  "RouterName": "whoami@docker",
+  "ServiceAddr": "10.0.2.6:80",
+  "ServiceName": "whoami@docker",
+  "ServiceURL": {
+    "Scheme": "http",
+    "Opaque": "",
+    "User": null,
+    "Host": "10.0.2.6:80",
+    "Path": "",
+    "RawPath": "",
+    "ForceQuery": false,
+    "RawQuery": "",
+    "Fragment": "",
+    "RawFragment": ""
+  },
+  "StartLocal": "2022-06-11T00:18:24.95226167Z",
+  "StartUTC": "2022-06-11T00:18:24.95226167Z",
+  "entryPointName": "web",
+  "level": "info",
+  "msg": "",
+  "time": "2022-06-11T00:18:24Z"
+}
+```
+
+## Risks
+
+> List the risks of this approach  
+> * There are always risks. What are the risks of this solution?  
+> * These are the things people will bring up in opposition to your idea or plans. Acknowledge them.  
+
+One concern we heard is what if someone deploys an application without baked in-app Keycloak integration and also skips Keycloak integration on the network level? 
+
+To control this kind of risk of not following the established practices, it might be good to look at how our deployment workflow is enforced.
+
+Another risk is what if the system capacity will not be sufficient to meet the production workloads?
+
+To control the kind of performance saturation risk, it is good to have an experienced vendor, think Traefik Labs, to support our OIDC integration starting from the trial period and system design time following through the production monitoring and necessary upgrades or patches over time. 
+
+Approximate load considerations:
+
+* We don't know yet for sure what the load on the OpenID-Connect (TraefikEE-to-Keycloak) would be, so it is good to have a vendor for enterprise-grade support, such as Traefik Labs.
+* We are targeting the new version of VETS-API that is coming to EKS, in some part as monolith, in some part as a collection of microserviecs. 
+* In the new system, in the spirit of ZTA, we expect some auth & authz activities between the microservices every time a user comes in with a new session plus a possibility of scheduled API calls w/o a user. 
+
+IET in-house research of the production monolith VETS-API can be found [here](https://docs.google.com/spreadsheets/d/14iNeWOyaMe0aqVtgfCqTE7cl3xAfQ7YJDsjRbADOXCk/edit#gid=1482742543). If you have no access to that google doc, here is the data sampling from Grafana, which might have some Lighthouse activities mixed in.
+
+Top 10 endpoints as of Winter 2022 sampling:
```

Comment:
This is end user (Veteran) traffic volumes correct?

Reply:
No, most of these endpoints calls resulted from different algorithms needed for VETS-API to function.